tt-train/tests/core/distributed_test.cpp

```diff
@@ -0,0 +1,144 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+
+#include <core/xtensor_all_includes.hpp>
+
+#include "core/distributed_mapping.hpp"
+
+template <typename T>
+class MeshOpsTest : public ::testing::Test {
+protected:
+    // Common setup could go here if needed
+};
+
+using TestTypes = ::testing::Types<std::uint32_t, float>;
+TYPED_TEST_SUITE(MeshOpsTest, TestTypes);
```

Comment:
@rfurko-tt ptal what cool feature I found.
@ayerofieiev-tt  you probably already know about it but still looks cool.